### PR TITLE
delete model chunks for other models except remote model

### PR DESCRIPTION
**Pull request by dhrubo-os**
### Description
[Describe what this change achieves]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dbwiddis — COMMENTED** [review]
Looks generally good for an approach to skipping deleting chunks... but.

Special-casing remote models is just adding a lot of code to bypass the real issue, which is that we're trying to delete non-existent chunks. This should be a "not found" situation, not an exception.

The error message is generated here:
```java
        } else if (!response.getBulkFailures().isEmpty()) {
            errorMessage = OS_STATUS_EXCEPTION_MESSAGE + ", " + BULK_FAILURE_MSG + modelId;
```
called in the else block from 
```java
            if ((r.getBulkFailures() == null || r.getBulkFailures().size() == 0)
                && (r.getSearchFailures() == null || r.getSearchFailures().size() == 0)) {
                log.debug(getErrorMessage("All model chunks are deleted for the provided model.", modelId, isHidden));
                actionListener.onResponse(true);
            } else {
                returnFailure(r, modelId, actionListener);
            }
```
So why would there be bulk failures if there's nothing to delete?  

There's exactly one bulk failure.  `getMessage()` returns:
> [.plugins-ml-model/g85FwXeGSLOy5CMEvIhWBg][[.plugins-ml-model][0]] VersionConflictEngineException[[5OaNTpEB-VCFMaCgxqtO]: version conflict, required seqNo [1], primary term [1]. but no document was found]

`.getSeqNo()` returns -2 (the unassigned one) and `getTerm()` returns 0 for primary term.  I don't know why the bulk request is strictly requiring values.

But it seems to me we could leave all the code as is and just make this change:
```java
        client.execute(DeleteByQueryAction.INSTANCE, deleteModelsRequest, ActionListener.wrap(r -> {
            if ((r.getBulkFailures() == null || r.getBulkFailures().isEmpty()
            // Ignore VersionConflict failure if document not found
                || r.getBulkFailures().stream().noneMatch(f -> f.getSeqNo() > SequenceNumbers.UNASSIGNED_SEQ_NO))
                && (r.getSearchFailures() == null || r.getSearchFailures().isEmpty())) {
                log.debug(getErrorMessage("All model chunks are deleted for the provided model.", modelId, isHidden));
                actionListener.onResponse(true);
            } else {
                returnFailure(r, modelId, actionListener);
            }
        }, e -> {
```
I did that locally on my Integ test PR and deletion works fine...

WDYT?

**Review by dbwiddis — COMMENTED** [review]
Dug into this a bit more, and there are some other possible fixes rather than excluding by sequence number.  Options include:
1. Use `refresh=wait_for` on the initial document deletion prior to the delete by query.  This would incur a bit of latency on model deletion but make sure the search hits didn't include it.
2. Modify the query on the chunk deletions to exclude the specific document we just deleted, so it's not included in the search hits.

File: plugin/src/test/java/org/opensearch/ml/action/models/DeleteModelTransportActionTests.java [hunk 1]
```diff
@@ -211,12 +211,12 @@ public void testDeleteRemoteModel_Success() throws IOException, InterruptedExcep
...
-            BulkByScrollResponse response = new BulkByScrollResponse(new ArrayList<>(), null);
-            listener.onResponse(response);
-            return null;
-        }).when(client).execute(any(), any(), any());
+//        doAnswer(invocation -> {
+//            ActionListener<BulkByScrollResponse> listener = invocation.getArgument(2);
+//            BulkByScrollResponse response = new BulkByScrollResponse(new ArrayList<>(), null);
+//            listener.onResponse(response);
+//            return null;
+//        }).when(client).execute(any(), any(), any());
```
[the remark this answers is not in the document]
Comment (dhrubo-os) [reply plugin/src/test/java/org/opensearch/ml/action/models/DeleteModelTransportActionTests.java:219]: I'll remove those.

**Comment by dhrubo-os** [comment]
>Special-casing remote models is just adding a lot of code to bypass the real issue, which is that we're trying to delete non-existent chunks. This should be a "not found" situation, not an exception.

Remote model doesn't have any model chunks. So I'm in support to reduce one unnecessary DB call for remote model. 

But that being said, I agree we should solve this issue about `This should be a "not found" situation, not an exception.`

**Comment by dbwiddis** [comment]
> But that being said, I agree we should solve this issue 

I think modifying the delete query to exclude the just-deleted document is the best solution here.

**Pull request merged**